tests/src/Kernel/CivicrmEntityTestBase.php

```diff
@@ -3,20 +3,24 @@
 namespace Drupal\Tests\civicrm_entity\Kernel;
 
 use Drupal\civicrm_entity\CiviCrmApiInterface;
-use Drupal\civicrm_entity\SupportedEntities;
+use Drupal\Core\Database\Database;
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\Core\DependencyInjection\ServiceModifierInterface;
 use Drupal\KernelTests\KernelTestBase;
 use Prophecy\Argument;
+use Prophecy\Promise;
 
 /**
  * Test base to aid in mocking the CiviCRM API.
  */
-abstract class CivicrmEntityTestBase extends KernelTestBase {
+abstract class CivicrmEntityTestBase extends KernelTestBase implements ServiceModifierInterface {
 
   /**
    * {@inheritdoc}
    */
   protected static $modules = [
     'system',
+    'user',
```

Comment:
We cannot use EntityKernelTestBase because it caused entity type manager to run and `SupportedEntities` accesses the database too early. And the `user.permissions` service is required.